### PR TITLE
added christmas tree patroling command

### PR DESCRIPTION
**Pull request by vhbfernandes**

### CODE REVIEW

**Comment by citypaul** [comment]
@vhbfernandes I just randomly found this PR - I suspect @mintuz didn't spot it at the time, as we get a lot of notifications in our job - I'm gonna give him a nudge to see if he'll merge this in for you :-)

**Comment by vhbfernandes** [comment]
Thanks @citypaul, this will be pretty useful again in a few months :smile:

File: commands/christmaspatrol.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    bb8.roll(Math.floor(distance/12),direction*i);
+    var chosen = (function(){
+      if (Math.random() > 0.5) {
+        return 1;
+       } else {
+        return 0;
+       }
+    })();
+    bb8.color(colors[chosen]);
+    i++;
```
Comment (mintuz) [inline commands/christmaspatrol.js]: Is it possible to give `i` a more descriptive name .

File: commands/christmaspatrol.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
+module.exports = function (bb8,options) {
+ var radius = options.radius; //mm
+ var distance = 2*Math.PI*radius;
+ var colors = ["red","green"];
+ var i = 0;
+ return setInterval(function() {
+    var direction = 30;
```
Comment (mintuz) [inline commands/christmaspatrol.js]: direction can be pulled out of the setInterval so it doesn't get redeclared every 500ms

File: commands/christmaspatrol.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+    bb8.roll(Math.floor(distance/12),direction*i);
+    var chosen = (function(){
+      if (Math.random() > 0.5) {
+        return 1;
+       } else {
+        return 0;
+       }
```
Comment (mintuz) [inline commands/christmaspatrol.js]: I'm not sure if you prefer this, but i typically write functions with conditional like this 

```
function() {
    
    if (Math.random() > 0.5) {
        return 1;
    }

    return 0;
}
```

**Review by mintuz — CHANGES_REQUESTED** [review]
Thanks @citypaul yep this completely missed my inbox with the notification spam from work 😄 Sorry I missed it @vhbfernandes it'll be a good addition for christmas 2018 though 🎄

File: commands/christmaspatrol.js (hunk 3 above)
Comment (citypaul) [reply commands/christmaspatrol.js]: couldn't this become: 
```js
function() {
  return Math.random() > 0.5 ? 1 : 0
}

// or using ES6 arrow syntax:
() => Math.random() > 0.5 ? 1 : 0
```

Comment (citypaul) [reply commands/christmaspatrol.js]: You could also change this logic entirely to just pull out a random element from the `colors` var. That way, if you wanted to add new colours in the future you'd be able to do it by just adding a new colour into the `colors` array.

You could do it like this:

```js
function getRandomElement(arr) {
   return arr[Math.floor(Math.random() * arr.length)];
}
```

Or just use `lodash` and cheat:

```js
// no need for a function as that's exactly what this does:
_.sample(colors)
```

Reference: https://lodash.com/docs/4.17.5#sample

In both instances, that means changing the `colors` var to this:

```js
var colors = ["green", "red", "yellow"];
```

Would start also randomly adding yellow in there too :-)

Personally I'd probably just use lodash.

**Comment by citypaul** [comment]
@vhbfernandes if you make the changes and @mintuz is happy to merge, don't forget to also add yourself in as a contributor in the main project README (there's a "contributors" section towards the end of the file).

Just do that as part of this PR 👍

File: commands/christmaspatrol.js (hunk 3 above)
Comment (mintuz) [reply commands/christmaspatrol.js]: Lodash is already part of the project as well so i've got no issue in using it. Nice suggestion though i'm not sure yellow is a christmas colour 😄

**Comment by vhbfernandes** [comment]
Hey guys, sorry for the delay, I've added the changes as requested. The code looks a lot cleaner now, thanks for the tips :)

File: commands/christmaspatrol.js (hunk 3 above)
Comment (vhbfernandes) [reply commands/christmaspatrol.js]: Also not sure about yellow, but great idea on using lodash, it looks way better now :+1:

File: commands/christmaspatrol.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+module.exports = function (bb8,options) {
+    var radius = options.radius; //mm
+    var distance = 2*Math.PI*radius;
+    var colors = ["red","green"];
+    var angle_counter = 0;
+    var direction = 30;
+    return setInterval(function() {
+        bb8.roll(Math.floor(distance/12),direction*angle_counter);
+        bb8.color(_.sample(colors));
```
Comment (mintuz) [inline commands/christmaspatrol.js:10]: at the top of the file you'll need `var _ = require('lodash')` for this to work, then it's good to go 😄

**Review by mintuz — CHANGES_REQUESTED** [review]

File: commands/christmaspatrol.js (hunk 4 above)
Comment (vhbfernandes) [reply commands/christmaspatrol.js:10]: Done, thanks!

**Comment by vhbfernandes** [comment]
Hey, let's remember this PR before xmas :)

**Review by citypaul — APPROVED** [review]
LGTM @mintuz let's get it merged in 🎄

**Review by mintuz — APPROVED** [review]

**Pull request merged**